packages/react-dom/src/__tests__/quoteAttributeValueForBrowser-test.js

```diff
@@ -9,40 +9,100 @@
+var ReactDOMServer;
+
+var ROOT_ATTRIBUTE_NAME;
+
 describe('quoteAttributeValueForBrowser', () => {
-  // TODO: can we express this test with only public API?
-  var quoteAttributeValueForBrowser = require('../shared/quoteAttributeValueForBrowser')
-    .default;
+  beforeEach(() => {
+    jest.resetModules();
+    React = require('react');
 
-  it('should escape boolean to string', () => {
-    expect(quoteAttributeValueForBrowser(true)).toBe('"true"');
-    expect(quoteAttributeValueForBrowser(false)).toBe('"false"');
+    ExecutionEnvironment = require('fbjs/lib/ExecutionEnvironment');
+    ExecutionEnvironment.canUseDOM = false;
+    ReactDOMServer = require('react-dom/server');
+
+    var DOMProperty = require('../shared/DOMProperty');
+    ROOT_ATTRIBUTE_NAME = DOMProperty.ROOT_ATTRIBUTE_NAME;
   });
 
-  it('should escape object to string', () => {
-    var escaped = quoteAttributeValueForBrowser({
-      toString: function() {
-        return 'ponys';
-      },
-    });
+  it('ampersand is escaped inside attributes', () => {
+    var response = ReactDOMServer.renderToString(<img data-attr="&" />);
+    expect(response).toMatch(
+      new RegExp(
+        '<img data-attr="&amp;" ' + ROOT_ATTRIBUTE_NAME + '=""' + '/>',
+      ),
+    );
+  });
 
-    expect(escaped).toBe('"ponys"');
+  it('double quote is escaped inside attributes', () => {
+    var response = ReactDOMServer.renderToString(<img data-attr={'"'} />);
+    expect(response).toMatch(
+      new RegExp(
+        '<img data-attr="&quot;" ' + ROOT_ATTRIBUTE_NAME + '=""' + '/>',
+      ),
+    );
   });
 
-  it('should escape number to string', () => {
-    expect(quoteAttributeValueForBrowser(42)).toBe('"42"');
+  it('single quote is escaped inside attributes', () => {
+    var response = ReactDOMServer.renderToString(<img data-attr="'" />);
+    expect(response).toMatch(
+      new RegExp(
+        '<img data-attr="&#x27;" ' + ROOT_ATTRIBUTE_NAME + '=""' + '/>',
+      ),
+    );
   });
 
-  it('should escape string', () => {
-    var escaped = quoteAttributeValueForBrowser(
-      '<script type=\'\' src=""></script>',
+  it('greater than entity is escaped inside attributes', () => {
+    var response = ReactDOMServer.renderToString(<img data-attr=">" />);
+    expect(response).toMatch(
+      new RegExp('<img data-attr="&gt;" ' + ROOT_ATTRIBUTE_NAME + '=""' + '/>'),
     );
-    expect(escaped).not.toContain('<');
-    expect(escaped).not.toContain('>');
-    expect(escaped).not.toContain("'");
-    expect(escaped.substr(1, -1)).not.toContain('"');
+  });
+
+  it('lower than entity is escaped inside attributes', () => {
+    var response = ReactDOMServer.renderToString(<img data-attr="<" />);
+    expect(response).toMatch(
+      new RegExp('<img data-attr="&lt;" ' + ROOT_ATTRIBUTE_NAME + '=""' + '/>'),
+    );
+  });
 
-    escaped = quoteAttributeValueForBrowser('&');
-    expect(escaped).toBe('"&amp;"');
+  it('number is escaped to string inside attributes', () => {
```

Comment:
@gaearon from here, tests for number, object and script tags on attributes have been added back. Above tests for text content in nodes contemplate these cases too, let me know if you ahve any more concerns or something that I might have missed.